javascript/code-challenges/challenges-05.test.js

```diff
@@ -0,0 +1,371 @@
+'use strict';
+
+const { last } = require("cheerio/lib/api/traversing");
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function that iterates over an array of people objects
+and creates a new list of each person's full name using the array method 'map'.
+Each object will have the shape {firstName:string, lastName:string}
+E.g. [ { firstName:"Jane", lastName:"Doe" }, { firstName:"James", lastName:"Bond"}]
+should convert to ["Jane Doe", "James Bond"]
+Note the space in between first and last names.
+You can assume that neither firstName nor lastName will be blank
+------------------------------------------------------------------------------------------------ */
+const toLastNames = people => {
+  return people.map(fullName => {
+    return fullName.firstName + " " + fullName.lastName;
+  });
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named addValues that, given an array of numbers as input, uses reduce to add the values in the array.
+
+------------------------------------------------------------------------------------------------ */
+
+const addValues = (arr) => {
+  return arr.reduce((arr, crr) => {
+    return arr = arr + crr;
+  },0);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named addPurchases that, given an array of objects as input, uses reduce to find the total amount purchased. Each object contains the keys `item` and `purchasePrice` like the example.
+
+{
+  item: 'switch'
+  purchasePrice: 399
+}
+
+------------------------------------------------------------------------------------------------ */
+
+const addPurchases = (arr) => {
+  return arr.reduce((firstItem, currentPrice) => {
+    return firstItem = firstItem + currentPrice.purchasePrice;
+  },0);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named countNumberOfElements that, given an array as input, uses reduce to count the number of elements in the array.
+
+Note: You may not use the array's built-in length property.
+------------------------------------------------------------------------------------------------ */
+
+const countNumberOfElements = (arr) => {
+  return arr.reduce((elementAmount, currNum) => {
+    return elementAmount = currNum;
+  },0);
+};
+
```

Comment:
Haha, so this doesn't actually work unless the test is 1, 2, 3, 4, 5. If the test ran on basically any of number set this would fail. Luckily for you the test is 1, 2, 3, 4, 5!
But to keep a tally, you'd want to do something like
```js
elementAmount += 1;
||
elementAmount = elementAmount + 1;
```